polkadot/runtime/common/src/impls.rs

```diff
@@ -98,13 +100,104 @@ pub fn era_payout(
 	(staking_payout, rest)
 }
 
+/// Versioned locatable asset type which contains both an XCM `location` and `asset_id` to identify
+/// an asset which exists on some chain.
+#[derive(
+	Encode, Decode, Eq, PartialEq, Clone, RuntimeDebug, scale_info::TypeInfo, MaxEncodedLen,
+)]
+pub enum VersionedLocatableAsset {
```

Comment:
I still don't see how this can't be replaced by `VersionedMultiAsset`.

Reply:
I think it's just more user friendly and gives more control for runtime over it's API. also for a very little cost.

an alternative will be 
``` rust 

pub enum VersionedLocatableAsset {
  /// The (relative) location in which the asset ID is meaningful.
  location: xcm::VersionedMultiLocation,
  /// The asset's ID.
  asset_id: xcm::VersionedAssetId,
}
```
+ test ensuring that a version which is still in use is not removed from versioned type, and if some removed, a migration will be required.

Reply:
we can even use `xcm::LocatableAssetId`, but every version upgrade will require a migration for treasury pallet